bentoml/_internal/bento/build_config.py

```diff
@@ -450,22 +525,28 @@ def write_to_bento(self, bento_fs: "FS", build_ctx: str) -> None:
 
         # write pip install args to a text file if applicable
         if pip_args:
-            with bento_fs.open(fs.path.join(py_folder, "pip_args.txt"), "w") as f:
+            with bento_fs.open(fs.path.combine(py_folder, "pip_args.txt"), "w") as f:
                 f.write(" ".join(pip_args))
 
-        if self.lock_packages:
+        if self.lock_packages:  # pragma: no cover
+            # This section is not covered by tests and we relies
+            # on jazzband/pip-tools test coverage
```

Comment:
we can still test this section by verifying the right pip-tools commands/APIs are invoked, also possible to cover it in integration or e2e tests.